packages/tabs/src/styled/StyledTab.ts

```diff
@@ -5,55 +5,102 @@
  * found at http://www.apache.org/licenses/LICENSE-2.0.
  */
 
-import PropTypes from 'prop-types';
-import styled from 'styled-components';
-import classNames from 'classnames';
-import TabStyles from '@zendeskgarden/css-tabs';
-import { retrieveComponentStyles } from '@zendeskgarden/react-theming';
+import styled, { DefaultTheme, css, ThemeProps } from 'styled-components';
+import { retrieveComponentStyles, getColor, DEFAULT_THEME } from '@zendeskgarden/react-theming';
 
 const COMPONENT_ID = 'tabs.tab';
 
 interface IStyledTabProps {
-  isHovered?: boolean;
-  isFocused?: boolean;
-  isActive?: boolean;
   disabled?: boolean;
   isSelected?: boolean;
 }
 
+/**
+ * A high specificity is needed to apply the border-color in vertical orientations
+ */
+const colorStyles = ({ theme, isSelected }: IStyledTabProps & ThemeProps<DefaultTheme>) => css`
+  color: ${isSelected ? getColor('primaryHue', 600, theme) : 'inherit'};
+
+  &&& {
+    border-color: ${isSelected && 'currentColor'};
+  }
+
+  &:hover {
+    color: ${props => getColor('primaryHue', 600, props.theme)};
+  }
+
+  &:focus {
+    outline: none;
+    text-decoration: none;
+  }
+
+  &:active {
+    border-color: currentColor;
+    color: ${props => getColor('primaryHue', 600, props.theme)};
+  }
+
+  &::before {
+    transition: box-shadow 0.1s ease-in-out;
+    content: '';
+  }
+
+  &:active::before {
+    box-shadow: none;
+  }
+`;
+
+const focusStyles = ({ theme }: ThemeProps<DefaultTheme>) => css`
+  &[data-garden-focus-visible] {
+    color: ${getColor('primaryHue', 600, theme)};
+  }
+
+  &[data-garden-focus-visible]::before {
+    position: absolute;
+    top: ${theme.space.base * 2.5}px;
+    right: ${theme.space.base * 6}px;
+    left: ${theme.space.base * 6}px;
+    border-radius: ${theme.borderRadii.md};
+    box-shadow: inset ${theme.shadows.sm(getColor('primaryHue', 600, theme, 0.35)!)};
```

Comment:
This would require us to have two areas that style the `::before` element on focus-visible. I found it harder to reason when focus styling is spaced around in multiple helper functions.

With this being one of the few components that focus a pseudo-element I'd prefer to keep this logic separate from the colors.

Reply:
These are now split.